hdfs-scheduler/src/main/java/org/apache/mesos/hdfs/scheduler/HdfsScheduler.java

```diff
@@ -127,6 +126,10 @@ public void statusUpdate(SchedulerDriver driver, TaskStatus status) {
       status.getMessage(),
       liveState.getStagingTasksSize()));
 
```

Comment:
Not convinced of the observer pattern here. I don't think we should notify observers every time we get a status update. This means observers will get notified everytime (not just during reconciliation). At least we could notify observers only when we are reconciling.